src/main/controllers/ChallengeController.kt

```diff
@@ -24,7 +24,7 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
     }
 
     override fun findOne(user: UserAccount, id: Int): SOAResult<Challenge> {
-        throw NotImplementedError()
+        return GetOneChallengeService.execute(id)
```

Comment:
seeing this now -- not sure the logic needs its own service...

might be better to just put 

val challengeResult = DaoService.execute {
            Challenge.findById(challengeId)
        }

in the controller and have the DaoService.throwOrReturn(
and return

Reply:
basically the logic for GetOneChallengeService is too simple to be a service really -- it's literally one query without any checks or anything -- justa . findbyid